lib/src/signed_video_h26x_auth.c

```diff
@@ -411,13 +407,9 @@ verify_hashes_with_hash_list(signed_video_t *self,
   if (latest_match_idx == -1) {
     DEBUG_LOG("Never found a matching hash at all");
     int num_missing_nalus = num_expected_hashes - num_invalid_nalus_since_latest_match;
-    // We do not know where in the sequence of NALUs they were lost. Simply add them before the
-    // first item. If the first item needs a second opinion, that is, it has already been verified
-    // once, we append that item. Otherwise, prepend it with missing items.
-    const bool append = !!nalu_list->first_item->nalu->is_first_nalu_in_gop;
     // No need to check the return value. A failure only affects the statistics. In the worst case
     // we may signal SV_AUTH_RESULT_OK instead of SV_AUTH_RESULT_OK_WITH_MISSING_INFO.
-    h26x_nalu_list_add_missing(nalu_list, num_missing_nalus, append, nalu_list->first_item);
+    h26x_nalu_list_add_missing(nalu_list, num_missing_nalus, true, nalu_list->first_item);
```

Comment:
I think we should add a comment like "TODO: Investigate if adding the missing items at the beginning of the list can cause any issues when validating multiple GOPs in one go."